libical-glib: Use libical's coding style in generated files

mcrha opened the pull request:
Follows https://github.com/libical/libical/pull/1116

tintou commented on src/libical-glib/i-cal-object.c.in:
```diff
@@ -341,8 +341,9 @@ i_cal_object_construct(GType object_type,
-    if (owner)
+    if (owner) {
         g_return_val_if_fail(G_IS_OBJECT(owner), NULL);
+    }
```
I've seen lot of occurrences of this in GLib projects:
```suggestion
    g_return_val_if_fail(!owner || G_IS_OBJECT(owner), NULL);
```

mcrha replied:
It's just a different way to check the same, with less output on terminal when not using the suggestion :)

mcrha commented:
Apart of getting plenty of errors about some included headers missing and some config.h claims from the
```
cd _build; clang-tidy --checks=readability-braces-around-statements `find . -name "*.c" -o -name "*.h"`
```
I see also:
```
.../_build/src/libical-glib/i-cal-object.h:43:31: warning: macro argument should be enclosed in parentheses [bugprone-macro-parentheses]
   43 |   G_GNUC_UNUSED static inline ModuleObjName * MODULE##_##OBJ_NAME (gpointer ptr) {                       \
      |                               ^
      |                               (            )
```

but it's a false positive, because when I enclose it into the parentheses, then the compiler yells at me (for each use of this macro):
```
.../_build/src/libical-glib/i-cal-attach.h:25:35: error: expected identifier or ‘(’ before ‘ICalAttach’
   25 | I_CAL_GLIB_DECLARE_DERIVABLE_TYPE(ICalAttach, i_cal_attach, I_CAL, ATTACH, ICalObject)
      |                                   ^~~~~~~~~~
.../_build/src/libical-glib/i-cal-object.h:43:32: note: in definition of macro ‘I_CAL_GLIB_DECLARE_DERIVABLE_TYPE’
   43 |   G_GNUC_UNUSED static inline (ModuleObjName) * MODULE##_##OBJ_NAME (gpointer ptr) {                     \
      |                                ^~~~~~~~~~~~~
```

otherwise the code should be fine. Let me know if you find anything else, please, I can include the changes for it here.

winterz commented:
try:

```
cd _build; clang-tidy --checks='-*,readability-braces-around-statements' `find . -name "*.c" -o -name "*.h"`
```

and you'll see a bunch more readability-braces-around-statements issues
for example in i-cal-timezone.c I see 12 such

mcrha commented:
> and you'll see a bunch more readability-braces-around-statements issues for example in i-cal-timezone.c I see 12 such

Not for me, I get plenty of:
```
.../_build/src/libical-glib/i-cal-array.h:10:2: error: "Only <libical-glib/libical-glib.h> can be included directly." [clang-diagnostic-error]
   10 | #error "Only <libical-glib/libical-glib.h> can be included directly."

.../_build/src/libical-glib/i-cal-component.h:16:10: error: 'libical-glib/i-cal-forward-declarations.h' file not found [clang-diagnostic-error]
   16 | #include <libical-glib/i-cal-forward-declarations.h>
```

regardless whether I compile with gcc or clang.  It does not seem to understand CFLAGS, where I tried to `-I<dir>` for those headers. I have clang-19.1.0 here.

mcrha commented:
> almost i-cal-vcard-time.c has a few

Hrm, not shown here for some reason. It's similar to `i-cal-time.xml`, thus I changed it blindly, and I also replaced tabs with four spaces.
